src/webgpu/api/operation/buffers/map_oom.spec.ts

```diff
@@ -125,15 +126,19 @@ an out-of-memory error if allocation fails.
       .combine('usage', kBufferUsages)
   )
   .fn(async t => {
-    const { usage, size } = t.params;
+    const { oom, usage, size } = t.params;
 
-    const buffer = t.expectGPUError('out-of-memory', () =>
-      t.device.createBuffer({ mappedAtCreation: true, size, usage })
+    const buffer = t.expectGPUError(
+      'out-of-memory',
+      () => t.device.createBuffer({ mappedAtCreation: true, size, usage }),
+      oom
     );
 
-    // Smaller range inside a too-big mapping
     const mapping = buffer.getMappedRange(0, 16);
     t.expect(mapping.byteLength === 16);
-    buffer.unmap();
-    t.expect(mapping.byteLength === 0, 'Mapping should be detached');
+
+    t.expectGPUError('validation', () => buffer.unmap(), oom);
```

Comment:
Weird that we get a validation error on unmap, and yet it has an effect (detaches the ArrayBuffer). I think we need to examine this corner case in the spec.

Reply:
Oh, actually I think the spec is already correct here:
https://gpuweb.github.io/gpuweb/#dom-gpubuffer-unmap

> Note: It is valid to unmap an error GPUBuffer that is mapped at creation because the Content timeline might not know it is an error GPUBuffer.

Incidentally, I think the spec is redundant, it shouldn't be possible for a buffer to be both mapped and destroyed.